### DISCUSSION TITLE
Generate waveform without mediaElement

### DISCUSSION THREAD

**Discussion by rangermanWSC** [katspaugh/wavesurfer.js#2677] [Q&A]
Hi, I followed the instructions on the FAQ page about creating the peak array on the server, but didn’t manage to draw the waveform without the audio file (I need the server to return the created json file only).
Is there a way to it? And also, if I don’t pass the audio file, how would wavesurfer know the size of the audio (in seconds to calculate the width)?

**Comment by ivanjaros** [comment]
I am going to be doing the same since i will be serving large files(many hours long). Can anyone recommend a Go library that can generate this? I saw a few but am not sure which to pick or whether I need to write my own.

**Comment by entonbiba** [comment]
I was working on a api that allows to generate the PCM data from audio tracks just haven't had much time to spin it up lately. Will share it here once it's available, however you can use something like this:

https://github.com/bbc/audiowaveform

Run this on the server:
``` 
audiowaveform -i audiofile.mp3 -o audiofile.json --pixels-per-second 20 --bits 8
```


Load the peaks json data on browser:
``` javascript
fetch('audiofile.json')
.then(response => {
    if (!response.ok) {
        throw new Error("HTTP error " + response.status);
    }
    return response.json();
})
.then(peaks => {
    console.log('loaded peaks! sample_rate: ' + peaks.sample_rate);

    // load peaks into wavesurfer.js
    wavesurfer.load(mediaElt, peaks.data);
})
.catch((e) => {
    console.error('error', e);
});
```

**Comment by katspaugh** [comment]
It's currently not possible to just render a waveform w/o audio, but this use case makes sense and we could add this option. PRs welcome. The duration can be calculated from the peaks if their sample rate is known. I believe the audiowaveform tool can do that.

**Reply by entonbiba** [reply]
I've been thinking of adding this feature, which essentially would generate a silent audio buffer as an option to work with simply rendering the peaks only. Not sure how requested this is though to spend time on it.

**Reply by katspaugh** [reply]
I think it's probably best in general if the person who actually needs this feature and will use and test it in their app works on it. :)
But if it tickles your fancy enough, doing it in the wavesurfer-ts is probably better than in the current version.

**Reply by entonbiba** [reply]
@katspaugh ah yes I'll pr on that might need collab access if anything once anything is available on my end ;)
